chore(docs): Azure App Insights Recipes (#2821)
kubeshop/tracetest: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md

- Reactively review application execution data to determine the cause of an incident.

[OpenTelemetry Collector Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib) - The official OpenTelemetry Distribution for packages outside of the core collector.

## Sample Node.js App with Azure App Insights, The OpenTelemetry Collector and Tracetest

This is a simple quick start guide on how to configure a Node.js app to use instrumentation with traces and Tracetest for enhancing your E2E and integration tests with trace-based testing. The infrastructure will use Azure App Insights as the trace data store, the OpenTelemetry Collector to process and route the telemetry data and a Node.js app to generate it.

## Prerequisites

You will need [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine to run this quick start app!

And the [App Insights Instrumentation Key](https://learn.microsoft.com/en-us/azure/bot-service/bot-service-resources-app-insights-keys?view=azure-bot-service-4.0) from your instance.

## Project Structure

The project is built with Docker Compose.

### 1. Node.js App

> The `Dockerfile` in the root directory is for the Node.js app.

Reviewer: ```suggestion
The `Dockerfile` and the `docker-compose.yaml` in the root directory are for the Node.js app.
```

### 2. Tracetest

> The `docker-compose.yaml` file, `tracetest.provision.yaml`, and `tracetest-config.yaml` in the `root` directory are for the setting up the Node.js App and Tracetest.

Reviewer: ```suggestion
The `docker-compose.yaml` file, `tracetest.provision.yaml`, and `tracetest-config.yaml` in the `tracetest` directory are for setting up the Node.js App and Tracetest.
```

### Docker Compose Network

All `services` in the `docker-compose.yaml` are on the same network and will be reachable by hostname from within other services.

## Node.js App

The Node.js app is a simple Express app, contained in the `src/index.js` file.

It is instrumented using the [Official OpenTelemetry Node.js SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node) wrapping the application code to send telemetry data to the OpenTelemetry Collector.

> The key following is the instrumentation section from the `src/tracing.js` file.

Reviewer 2: ```suggestion
The following key is the instrumentation section from the `src/tracing.js` file.
```
Is 'key' the correct word or should it be 'code'?

Reviewer: I'd say we should use "code".
